### PR TITLE
Enhance Job Messages in Datasets Details page --DRAFT

### PR DESCRIPTION
**Pull request by hujambo-dunia**
UPDATE: Relocated from this Galaxy main's repo: #[17154](https://github.com/galaxyproject/galaxy/pull/17154) to point to base branch's repo which more clearly indicates the changes made by author. Still reading comments on this PR, but closing to prevent from being merged. (For comments to original @bernt-matthias branch, visit this PR #[17016](https://github.com/galaxyproject/galaxy/pull/17016) - Cc: @jmchilton.)

### CODE REVIEW

File: lib/galaxy/tool_util/output_checker.py [hunk 1]
```diff
@@ -17,7 +29,9 @@ class DETECTED_JOB_STATE(str, Enum):
 ERROR_PEEK_SIZE = 2000
 
 
-def check_output_regex(job_id_tag, regex, stream, stream_name, job_messages, max_error_level):
+def check_output_regex(
+    regex: "ToolStdioRegex", stream: str, stream_name: str, job_messages: List[Dict[str, Any]], max_error_level: int
+) -> int:
```
Comment (jmchilton) [inline lib/galaxy/tool_util/output_checker.py:34]: ✅ I cannot find any place in external code where this is used so I think the dropping of the job_id_tag parameter here is fine.

**Comment by hujambo-dunia** [comment]
@bernt-matthias @mvdbeek Can I assume the `messages` datatype will always be an Object from the API? Or could be Null/String/Numeric? Relates to [assumption referenced here](https://github.com/galaxyproject/galaxy/pull/17154/files#diff-93ab7b000c2174f5dafe357a79b162b6d1d52c0bb5510e1aab90e391cee890d1R59).

**Comment by hujambo-dunia** [comment]
> > add @hujambo-dunia as a collaborator
> 
> Could do this if wanted. Just ping me.

@bernt-matthias Yes please - could you make the change @jmchilton recommends in your feature branch? Then, I can pull your changes into my branch.
     Also, could you please [add me as a collaborator to your repo](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository), so I can point this/my PR to your [feature branch PR](https://github.com/galaxyproject/galaxy/pull/17016/commits)?

**Comment by bernt-matthias** [comment]
>  Yes please - could you make the change @jmchilton recommends in your feature branch?

Could you remind me what this was?

**Comment by hujambo-dunia** [comment]
> > Yes please - could you make the change @jmchilton recommends in your feature branch?
> 
> Could you remind me what this was?

Nevermind, @jmchilton [marked the issue](https://github.com/galaxyproject/galaxy/pull/17154/files/7386f8a2bdd2de7a39616b1f371751c59aaebc65) as `resolved` on this PR.

**Pull request closed without merging**